gemrb/core/System/String.cpp

```diff
@@ -109,10 +109,12 @@ String* StringFromCString(const char* string)
 
 char* MBCStringFromString(const String& string)
 {
-	char* cStr = (char*)malloc(string.length()+1);
+	size_t allocatedBytes = string.length() * sizeof(String::value_type);
+	char *cStr = (char*)malloc(allocatedBytes);
```

Comment:
Thanks, forgot to check before. Why bother with the realloc though — can't we reserve enough in advance? Looks like it would also kill a FIXME.